tests/test_caching.py

```diff
@@ -83,3 +83,36 @@ def foo(): "deprecated since 1999"
                 with contextlib.redirect_stdout(buf):
                     run()
                 self.assertEqual(buf.getvalue(), ref)
+    
+    def test_set_cache(self):
+        fid, tmppy = tempfile.mkstemp(suffix='.py')
```

Comment:
please use `tempfile.NamedTemporaryFile` with `delete=False` here; Seeing calls to `os.close` makes me hitch ;-)